tools/Voron.Recovery/Recovery.cs

```diff
@@ -38,162 +43,198 @@ private long GetFilePosition(long offset, byte* position)
         public RecoveryStatus Execute(CancellationToken ct)
         {            
             var sw = new Stopwatch();
+            StorageEnvironment se = null;
             sw.Start();
-            var mem = Pager.AcquirePagePointer(null, 0);
-            long startOffest = (long)mem;
-            var fi = new FileInfo(_datafile);
-            var fileSize = fi.Length;
-            //making sure eof is page aligned
-            var eof = mem + (fileSize / _pageSize) * _pageSize; 
-            DateTime lastProgressReport = DateTime.MinValue;
-            using (var destinationStream = File.OpenWrite(_output))
-            using (var logFile = File.CreateText(Path.Combine(Path.GetDirectoryName(_output), LogFileName)))
-            using (var gZipStream = new GZipStream(destinationStream, CompressionMode.Compress, true))
-            using (var context = new JsonOperationContext(_initialContextSize, _initialContextLongLivedSize))
-            using (var writer = new BlittableJsonTextWriter(context, gZipStream))
+            if (_copyOnWrite)
             {
-                WriteSmugglerHeader(writer);
-                while (mem < eof)
+                Console.WriteLine("Recovering journal files, this may take a while...");
+                try
+                {
+
+                    se = new StorageEnvironment(_option);
+                    Console.WriteLine($"Journal recovery has completed successfully within {sw.Elapsed.TotalSeconds:N1} seconds");                    
+                }
+                catch (Exception e)
+                {
+                    Console.WriteLine($"Journal recovery failed, reason:{Environment.NewLine}{e}");
+                    // The reason i create a new storage enviroment option is because the old one is using copy on write
+                    // And it may have wrote junk to the datafile and i need to open a "clean" MMF of the data file.
+                    _option = StorageEnvironmentOptions.ForPath(Path.GetDirectoryName(_datafile));
```

Comment:
I have changed the code and forgot to take this into account.
The problem i was having is that i can't dispose of the storage environment because it disposes of the data pager.
i'm going to change all of this behavior.